lib/matplotlib/axes/_base.py

```diff
@@ -2013,7 +2018,7 @@ def draw(self, renderer=None, inframe=False):
         # frame in the foreground. Do this before drawing the axis
         # objects so that the spine has the opportunity to update them.
         if not (self.axison and self._frameon):
-            for spine in six.itervalues(self.spines):
+            for spine in _sorted_itervalues(self.spines):
```

Comment:
This is looping to over items to remove them from the artist list, this is not needed